### PR TITLE
BigQuery plugin failed to read credential

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Before:
1. Failed to use parameter in SQL statement
2. BQ client failed to read credentials

image

image

After:
Check flytesnack example: https://github.com/flyteorg/flytesnacks/pull/521

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
1. UseLegacySql should always be false
2. We don't need to pass a token source to the BQ client when using the default token source factory. 

`GetTokenSource` will try to create credentials by reading `$GOOGLE_APPLICATION_CREDENTIALS`, but by default the BQ client will read `$GOOGLE_APPLICATION_CREDENTIALS` when it creates a connection. Therefore, we don't need to create credentials by ourselves.
https://github.com/flyteorg/flyteplugins/blob/fea5d848a5fe20b30d3a1d24c9e18f3706a6b7c4/go/tasks/pluginmachinery/google/default_token_source_factory.go#L12-L14

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/plugins/webapi/bigquery/plugin.go [hunk 1]
```diff
@@ -456,7 +457,7 @@ func (p Plugin) newBigQueryClient(ctx context.Context, identity google.Identity)
 		options = append(options,
 			option.WithEndpoint(p.cfg.bigQueryEndpoint),
 			option.WithTokenSource(oauth2.StaticTokenSource(&oauth2.Token{})))
-	} else {
+	} else if p.cfg.GoogleTokenSource.Type != "default" {
```
Comment (kumare3) [inline go/tasks/plugins/webapi/bigquery/plugin.go:460]: is the else part use no token source? Can we log that?

**Review by kumare3 — COMMENTED** [review]
lgtm, one comment

File: go/tasks/plugins/webapi/bigquery/plugin.go (hunk 1 above)
Comment (pingsutw) [reply go/tasks/plugins/webapi/bigquery/plugin.go:460]: yeah.
BQ client will read env `GOOGLE_APPLICATION_CREDENTIALS` by default

**Review by EngHabu — APPROVED** [review]
